Add more LayoutUnit API tests for LayoutUnit::fromFloat*() behavior

Ahmad-S792 opened the pull request:
#### 0058f6b04bfbf152643d9cd7548dd10c197a176a
<pre>
Add more LayoutUnit API tests for LayoutUnit::fromFloat*() behavior

<a href="https://bugs.webkit.org/show_bug.cgi?id=266950">https://bugs.webkit.org/show_bug.cgi?id=266950</a>

Reviewed by Alan Baradlay.

Partial Merge: <a href="https://chromium-review.googlesource.com/c/chromium/src/+/4652129">https://chromium-review.googlesource.com/c/chromium/src/+/4652129</a>

This PR  imports additional API tests around LayoutUnit::fromFloat*() from Blink / Chromium.

NOTE: This PR does not add `quiet_NaN()` tests.

* Tools/TestWebKitAPI/Tests/WebCore/LayoutUnitTests.cpp:
(LayoutUnitFloat):
(FromFloatCeil):
(FromFloatFloor):
(FromFloatRound):

Canonical link: <a href="https://commits.webkit.org/272707@main">https://commits.webkit.org/272707@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/78719f76a3660f1353db9a27d4a1ca123edefeba

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/32796 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/11547 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/34657 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/35408 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/29625 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/33723 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/13891 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/8731 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/35408 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/33239 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/9736 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/29286 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/35408 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/8617 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/10/builds/29241 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/36738 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/29794 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/29645 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/34735 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/8739 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/6704 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/32604 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/10426 "Built successfully") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/9349 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/4225 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/9357 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas commented on Tools/TestWebKitAPI/Tests/WebCore/LayoutUnitTests.cpp:
```diff
@@ -266,4 +268,53 @@ TEST(WebCoreLayoutUnit, LayoutUnitUnaryMinus)
+
+TEST(WebCoreLayoutUnit, FromFloatRound)
+{
+    const float tolerance = 1.0f / kFixedPointDenominator;
```
We're defining `tolerance` thrice to the same value each time... maybe just pull it out into an anonymous namespace?

```cpp
namespace TestWebKitAPI {

namespace {
constexpr float tolerance = 1.0f / kFixedPointDenominator;
}

TEST(WebCoreLayoutUnit, LayoutUnitInt)
{
```

Ahmad-S792 replied:
Good idea!

Ahmad-S792 replied:
Done in [c3ae314](https://github.com/WebKit/WebKit/commit/c3ae31426987322632c96eee6b55505e3314404a).

aprotyas reviewed (commented):
Looks good to me, with one question...

> NOTE: This PR does not add `quiet_NaN()` tests.

Why?

aprotyas commented on Tools/TestWebKitAPI/Tests/WebCore/LayoutUnitTests.cpp:
```diff
@@ -36,6 +36,10 @@ using namespace WebCore;
 namespace TestWebKitAPI {
 
+namespace {
+    constexpr float tolerance = 1.0f / kFixedPointDenominator;
```
You can drop the indent, so that we comply with https://webkit.org/code-style-guidelines/#indentation-namespace

Ahmad-S792 replied:
Thanks! For some reason style checker didn't complain. Surprised!

Ahmad-S792 replied:
[Done](https://github.com/WebKit/WebKit/commit/78719f76a3660f1353db9a27d4a1ca123edefeba).

aprotyas reviewed (commented):
LGTM with a style comment.

Ahmad-S792 committed:
Add more LayoutUnit API tests for LayoutUnit::fromFloat*() behavior

https://bugs.webkit.org/show_bug.cgi?id=266950

Reviewed by Alan Baradlay.

Partial Merge: https://chromium-review.googlesource.com/c/chromium/src/+/4652129

This PR  imports additional API tests around LayoutUnit::fromFloat*() from Blink / Chromium.

NOTE: This PR does not add `quiet_NaN()` tests.

* Tools/TestWebKitAPI/Tests/WebCore/LayoutUnitTests.cpp:
(LayoutUnitFloat):
(FromFloatCeil):
(FromFloatFloor):
(FromFloatRound):

Canonical link: https://commits.webkit.org/272707@main
